message.d.ts

```diff
@@ -0,0 +1,588 @@
+import '../closure-library/closure/goog/array/array.js';
+import { BinaryReader } from './binary/reader.js';
+import { BinaryWriter } from './binary/writer.js';
+import { Map } from './map.js';
+/**
+ * Stores information for a single extension field.
+ *
+ * For example, an extension field defined like so:
+ *
+ * extend BaseMessage {
+ * optional MyMessage my_field = 123;
+ * }
+ *
+ * will result in an ExtensionFieldInfo object with these properties:
+ *
+ * {
+ * fieldIndex: 123,
+ * fieldName: {my_field_renamed: 0},
+ * ctor: proto.example.MyMessage,
+ * toObjectFn: proto.example.MyMessage.toObject,
+ * isRepeated: 0
+ * }
+ *
+ * We include `toObjectFn` to allow the JSCompiler to perform dead-code removal
+ * on unused toObject() methods.
+ *
+ * If an extension field is primitive, ctor and toObjectFn will be null.
+ * isRepeated should be 0 or 1.
+ *
+ * binary{Reader,Writer}Fn and (if message type) binaryMessageSerializeFn are
+ * always provided. binaryReaderFn and binaryWriterFn are references to the
+ * appropriate methods on BinaryReader/BinaryWriter to read/write the value of
+ * this extension, and binaryMessageSerializeFn is a reference to the message
+ * class's .serializeBinary method, if available.
+ */
+export declare class ExtensionFieldInfo<T = any> {
+    /**
+     * Stores information for a single extension field.
+     *
+     * For example, an extension field defined like so:
+     *
+     * extend BaseMessage {
+     * optional MyMessage my_field = 123;
+     * }
+     *
+     * will result in an ExtensionFieldInfo object with these properties:
+     *
+     * {
+     * fieldIndex: 123,
+     * fieldName: {my_field_renamed: 0},
+     * ctor: proto.example.MyMessage,
+     * toObjectFn: proto.example.MyMessage.toObject,
+     * isRepeated: 0
+     * }
+     *
+     * We include `toObjectFn` to allow the JSCompiler to perform dead-code removal
+     * on unused toObject() methods.
+     *
+     * If an extension field is primitive, ctor and toObjectFn will be null.
+     * isRepeated should be 0 or 1.
+     *
+     * binary{Reader,Writer}Fn and (if message type) binaryMessageSerializeFn are
+     * always provided. binaryReaderFn and binaryWriterFn are references to the
+     * appropriate methods on BinaryReader/BinaryWriter to read/write the value of
+     * this extension, and binaryMessageSerializeFn is a reference to the message
+     * class's .serializeBinary method, if available.
+     * @param fieldName This has the extension field name as a property.
+     */
+    constructor(fieldNumber: number, fieldName: object | null, ctor: {
+        new (): Message | null;
+    }, toObjectFn: (() => object) | null, isRepeated: number);
+    private noStructuralTyping____protobuf_javascript_message_ExtensionFieldInfo;
+    fieldIndex: any;
```

Comment:
These `any` types should probably be restricted to match the constructor arguments. e.g.

```ts
fieldIndex: number;
```

for this one.